src/types.jl

```diff
@@ -1210,9 +1218,20 @@ function ^(a::SN, b)
     elseif b isa Number && b < 0
         Div(1, a ^ (-b))
     elseif ismul(a) && b isa Number
-        coeff = unstable_pow(a.coeff, b)
+        new_dict = mapvalues((k, v) -> b*v, a.dict)
+        if b isa Rational
+            if isinteger(b)
+                coeff = a.coeff ^ numerator(b)
+            else
+                coeff = 1
+                merge!(new_dict, Dict(term(^, a.coeff, b) => 1))
```

Comment:
Constructing a `Dict` is unnecessarily expensive here, as it allocates vectors of length 16.
https://github.com/JuliaLang/julia/blob/6f3fdf7b36250fb95f512a2b927ad2518c07d2b5/base/dict.jl#L68-L71